frontend/components/FilterSearch.js

```diff
@@ -0,0 +1,153 @@
+import { useState, useEffect } from 'react'
+import s from 'styled-components'
+import Select from 'react-select/async'
+import fuzzysort from 'fuzzysort'
+
+import { Icon } from './common'
+import { FilterHeader } from './DropdownFilter'
+import {
+  mediaMaxWidth,
+  MD,
+  ANIMATION_DURATION,
+} from '../constants/measurements'
+import {
+  ALLBIRDS_GRAY,
+  MEDIUM_GRAY,
+  CLUBS_RED,
+  CLUBS_GREY,
+  FOCUS_GRAY,
+  BORDER,
+  WHITE,
+} from '../constants/colors'
+
+const SearchWrapper = s.div`
+  margin-bottom: 30px;
+  max-height: 0;
+  opacity: 0;
+  overflow: visible;
+  transition: all ${ANIMATION_DURATION}ms ease;
+
+  ${({ active }) => active && 'max-height: 150vh; opacity: 1;'}
+
+  ${mediaMaxWidth(MD)} {
+    height: auto;
+    overflow: visible;
+    width: 100%;
+    margin: 0;
+    padding: 8px 1rem;
+    border-bottom: 1px solid ${BORDER};
+    position: fixed;
+    left: 0;
+    z-index: 1000;
+    background: ${WHITE};
+    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.075);
+  }
+`
+
+const SearchIcon = s.span`
+  cursor: pointer;
+  color: ${MEDIUM_GRAY};
+  opacity: 0.5;
+  margin-right: 6px;
+
+  ${mediaMaxWidth(MD)} {
+    right: 24px;
+  }
+`
+
+const Search = ({ searchTags, recommendedTags, updateTag }) => {
+  // Custom styles for the react-select
+  const styles = {
+    control: ({ background, ...base }, { isFocused, isSelected }) => {
+      const isEmphasized = isFocused || isSelected
+      return {
+        ...base,
+        border: `1px solid ${ALLBIRDS_GRAY}`,
+        background: isEmphasized ? FOCUS_GRAY : background,
+        boxShadow: 'none',
+        '&:hover': {
+          background: FOCUS_GRAY,
+        },
+      }
+    },
+    option: ({ background, ...base }, { isFocused, isSelected }) => {
+      const isEmphasized = isFocused || isSelected
+      return {
+        ...base,
+        background: isEmphasized ? FOCUS_GRAY : background,
+        color: CLUBS_GREY,
+      }
+    },
+  }
+
+  // Overriding specific components of the react-select
+  const components = {
+    IndicatorSeparator: () => null,
+    DropdownIndicator: () => <SearchIcon><Icon name="tag" alt="" /></SearchIcon>,
+  }
+
+  return <Select
+    isMulti
+    styles={styles}
+    components={components}
+    loadOptions={searchTags}
+    defaultOptions={recommendedTags}
+    cacheOptions
+    onChange={(_, selectEvent) => {
+      const { action, option } = selectEvent
+      action === 'select-option' && updateTag(option, 'Tags')
```

Comment:
What does this do?

Reply:
this onChange is provided by react-select, it handles when an option is selected. I check that the event is "select" here and then add the tag to the selected tags using updateTag